src/ProxyHost.ts

```diff
@@ -137,6 +140,18 @@ export default class ProxyHost {
     this.startingHost = false;
   }
 
+  public async isContainerRunning():Promise<boolean> {
+    let interval:NodeJS.Timer;
+    return new Promise((resolve) => {
+      interval = setInterval(async () => {
+        if (this.containerRunning) {
+          resolve(true);
+          clearInterval(interval);
+        }
+      }, 250);
+    });
+  }
```

Comment:
This is not guaranteed to always work in my opinion. It might fail if the containers webserver needs some time to start and isn't ready yet to accept a connection. 
This is the reason we have used requests & status code validations before redirecting.
What do you think?

Reply:
I use the containerRunning because it's set to true in the checkContainerReady method (see https://github.com/ItsEcholot/ContainerNursery/blob/main/src/ProxyHost.ts#L161), so I think it's appropriate to use it. Additionally, I've been running some tests with my container, and it works fine for me. let me know!

Reply:
Sorry I missed this reply.
`containerRunning` does indeed guarantee that the container is running, **but** this does not guarantee that the application inside the container (in our case most likely a webserver of some sorts) has started up and is **ready** to accept connections.
While this might work fine for a lot of containers it is in my opinion brittle.
You would need to implement an actual request check (as we already do [on the waiting page](https://github.com/ItsEcholot/ContainerNursery/blob/cf98c9748d216446a8d6065328aec0522a1a1581/views/placeholder.ejs#L108)) to ensure the webserver is ready for our request.